tests/test_unittest/test_cli.py

```diff
@@ -172,47 +137,59 @@ def test_backend_choice(backend_name):
     assert backend_name in cli.BackendChoice.get_backend_names()
 
 
-def test_unenabled_backend_choice(clirunner, nocli_backend, simple_stencil):
+def test_gen_unenabled_backend_choice(clirunner, nocli_backend, simple_stencil, tmp_path):
     """
     Test the --backend option when an unenabled backend name is passed.
-
-    The :py:func:`nocli_backend` fixture temporarily injects a dummy backend named "nocli", which
-    is not CLI enabled (and provides no functionality in fact).
-
-    Actions:
-
-        .. code-block:: bash
-
-            $ gtpyc --backend=nocli <simple_stencil_path>
-
-    Outcome:
-
-        An error message warns the user that the chosen backend is not enabled for CLI.
-        The command aborts with error code 2.
-
     """
-    result = clirunner.invoke(cli.gtpyc, ["--backend=nocli", str(simple_stencil.absolute())])
+    result = clirunner.invoke(
+        cli.gtpyc,
+        [
+            "gen",
+            "--backend=nocli",
+            str(simple_stencil.absolute()),
+            "--output-path",
+            str(tmp_path / "test_gen_unenabled"),
+        ],
+    )
 
     assert result.exit_code == 2
     assert re.findall(r".*Backend is not CLI-enabled\..*", result.output)
 
 
-def test_enabled_backend_choice(clirunner, simple_stencil, backend_name, backend_enabled):
+def test_gen_enabled_backend_choice(
+    clirunner, simple_stencil, backend_name, backend_enabled, tmp_path
+):
     """
     Test an enabled backend.
```

Comment:
```suggestion
    Test if cli runs successfully or fails depending on a backend being enabled.
```
I think this test nullifies the usefulness of the previous test.

Reply:
I don't quite understand. The previous test ensures you get a helpful error message if you use an unenabled backend. This tests that you don't get an error when you use an enabled backend.

I think what you mean is that the parametrization should be narrowed to only run these tests on enabled / unenabled backends.

Reply:
Or I could integrate the error message check from the previous one into this one (probably better)

Reply:
Now I remember why I did this. It possible that at some point there is no "unenabled" backend in gt4py. The functionality of rejecting unenabled ones must still be tested in that case. So the most sensible course of action is to simply remove the extraneous `else` clause in this test case.